### PR TITLE
Refactor SimplePSF fitting procedure

### PR DESCRIPTION
**Pull request by rmjarvis**
When working on implementing the composite PSF models, I've realized that various aspects of the current code should be refactored or adjusted in various ways.  So this is the first in what I expect to be several PRs with changes that are geared toward making the Sum and Convolve PSF classes easier to implement.

This PR mostly involved breaking up the fitting iteration into separate functions.  This will for instance allow the composite classes to call these functions for each one of their components as part of their implementation of these steps.  That would not have been possible with the old way the code was structured.

### CODE REVIEW

**Review by theoschutt — APPROVED** [review]
Looks good to me! The way functions are split between the PSF and derived classes makes sense to me.

File: piff/config.py [hunk 1]
```diff
@@ -137,9 +137,6 @@ def process(config, logger=None):
     objects, wcs, pointing = Input.process(config['input'], logger=logger)
     stars = Select.process(config.get('select',{}), objects, logger=logger)
 
-    if len(stars) == 0:
```
[the remark this answers is not in the document]
Comment (rmjarvis) [reply piff/config.py]: Ah, you're looking at individual commits.  :)  I added this and then removed it when I realized it was unnecessary.  If you look at the net diff (the "Files changed" tab) it's not there.

**Comment by rmjarvis** [comment]
Thanks Theo!

File: piff/config.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (theoschutt) [reply piff/config.py]: Oh, thank you, that tab makes things simpler! haha

**Pull request merged**